docs/broker/filtering.md

```diff
@@ -0,0 +1,360 @@
+    advertise a single CEL expression composed of the union of its Triggers'
+    filters, allowing upstream sources to filter events before publishing to the
+    Broker.
+
+CEL is
+[memory-safe, side-effect-free, and terminating](https://github.com/google/cel-spec/blob/master/doc/langdef.md#overview),
+making it safe and secure to embed in a multi-tenant dispatcher process.
+
+## Examples
+
+### Single attribute exact match
+
+Specified with the SourceAndType field:
+
+```yaml
+spec:
+  filter:
+    sourceAndType:
+      type: com.github.pull.create
+```
+
+Specified with a CEL expression:
+
+```yaml
+spec:
+  filter:
+    cel:
+      expression: ce.type == "com.github.pull.create"
+```
+
+### Multiple attributes exact match
+
+Specified with the SourceAndType field:
+
+```yaml
+spec:
+  filter:
+    sourceAndType:
+      type: com.github.pull.create
+      source: https://github.com/knative/eventing/pulls/123
+```
+
+Specified with a CEL expression:
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    cel:
+      expression: >
+        ce.type == "com.github.pull.create" &&
+        ce.source == "https://github.com/knative/eventing/pulls/123"
+```
+
+### Prefix match on source
+
+Cannot be specified with the SourceAndType field.
+
+CEL doesn't support prefix match directly, but prefix match can be accomplished
+with a regular expression:
+
+```yaml
+spec:
+  filter:
+    cel:
+      expression: ce.source.matches("^https://github.com")
+```
+
+### Complex boolean expression
+
+Cannot be specified with the SourceAndType field.
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    cel:
+      expression: >
+        ce.type == "com.github.pull.create" ||
+        (ce.type == "com.github.issue.create" && ce.source.matches("proposals")
+
+```
+
+### Parsed extensions
+
+Cannot be specified with the SourceAndType field.
+
+_This example assumes the repository name is available as a CloudEvents
+extension with name `repository`._
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    cel:
+      parseExtensions: true
+      expression: >
+        ce.type == "com.github.pull.create" ||
+         (ce.type == "com.github.issue.create" &&
+          ce.repository == "proposals")
+
+```
+
+### Parsed data
+
+Cannot be specified with the SourceAndType field.
+
+_The `>` syntax is a standard yaml multiline string._
+
+_Until [google/cel-go#203](https://github.com/google/cel-go/issues/203) is
+resolved, dynamic integer fields in CEL must be compared as floats._
+
+```yaml
+spec:
+  filter:
+    cel:
+      parseData: true
+      expression: >
+        ce.type == "dev.knative.observation" &&
+        data.latency > 300.0
+```
+
+## Caveats
+
+### CEL
+
+The main caveat is the choice of CEL as expression language.
+
+CEL is new and the language is not officially supported by Google. Participation
```

Comment:
CEL is used within a number of production products at Google, so I think the docs need to be updated to reflect this.  It's more like a 'supported project' rather than a 'supported product' which is sort of splitting hairs, but an important distinction.